CardPayments/src/main/java/com/paypal/android/cardpayments/CardClient.kt

```diff
@@ -160,26 +160,35 @@ class CardClient internal constructor(
 
     private fun handleBrowserSwitchSuccess(browserSwitchResult: BrowserSwitchResult) {
         ApproveOrderMetadata.fromJSON(browserSwitchResult.requestMetadata)?.let { metadata ->
-            CoroutineScope(dispatcher).launch(approveOrderExceptionHandler) {
-                try {
-                    analyticsService.sendAnalyticsEvent(
-                        "card-payments:3ds:get-order-info:succeeded",
-                        metadata.orderId
-                    )
-                    val deepLinkUrl = browserSwitchResult.deepLinkUrl
-                    val result = CardResult(metadata.orderId, deepLinkUrl)
-                    notifyApproveOrderSuccess(result)
-                } catch (error: PayPalSDKError) {
-                    analyticsService.sendAnalyticsEvent(
-                        "card-payments:3ds:get-order-info:failed",
-                        metadata.orderId
-                    )
-                    throw error
-                }
+            try {
+                val deepLinkUrl = browserSwitchResult.deepLinkUrl
+                val result = parseApproveOrderDeepLink(metadata.orderId, deepLinkUrl)
+                notifyApproveOrderSuccess(result)
+            } catch (error: PayPalSDKError) {
+                analyticsService.sendAnalyticsEvent(
+                    "card-payments:3ds:get-order-info:failed",
+                    metadata.orderId
+                )
+                notifyApproveOrderFailure(error)
             }
         }
     }
 
+    @Throws(PayPalSDKError::class)
+    private fun parseApproveOrderDeepLink(orderId: String, deepLinkUrl: Uri?): CardResult {
+        if (deepLinkUrl == null) {
+            throw CardError.threeDSVerificationError
+        }
+
+        val error = deepLinkUrl.getQueryParameter("error")
+        if (error != null) {
+            throw CardError.threeDSVerificationError
+        } else {
+            val liabilityShift = deepLinkUrl.getQueryParameter("liability_shift")
+            return CardResult(orderId, deepLinkUrl, liabilityShift)
+        }
```

Comment:
Steven, we also check for conditions for success and throw another error if it is not in the format specified.
https://github.com/paypal/paypal-ios/blob/2896471c1533de72880afd18545a33a374f9cfc0/Sources/CardPayments/CardClient.swift#L140
What do you think?

Reply:
I can add those checks in 👍 . I missed that part from the iOS PR.

Reply:
Added here in [af7cbad](https://github.com/paypal/paypal-android/pull/227/commits/af7cbad6aede5a940ae13cc34ed4fd1bc06bb05f).